src/ConnectedMode/Migration/IMigrationPrompt.cs

```diff
@@ -101,5 +112,10 @@ private string GetSolutionPath()
 
             return fullSolutionName as string;
         }
+
+        public void Dispose()
+        {
+            migrationWizardController.MigrationWizardFinished -= OnMigrationWIzardFinished;
+        }
```

Comment:
It seems a bit odd to have separate `Clear` and `Dispose` methods (or perhaps it seems odd that `Dispose` doesn't call `Clear`).

If this class is `Shared` it won't be disposed until VS shuts down, but it's still neater to explicitly unregister the event handler.

Reply:
Considering we will change this to be NonShared Is it okay that for now I just added Clear into the Dispose()?